### PR TITLE
Pydantic Datum class

### PR DESCRIPTION
**Pull request by loriab**
Converts `qcel.Datum` to Pydantic model. Changes
  * (a) comment, doi, glossary fields must be accessed by keyword,
  * (b) `to_dict()` becomes `dict()` and instead of label, units, data only, now comment, doi, glossary present _if_ non-default,
  * (c) complex values no longer list-ified by `to_dict()`.

### CODE REVIEW

File: qcelemental/datum.py [hunk 1]
```diff
@@ -28,8 +29,35 @@ class Datum(collections.namedtuple('Datum', 'label units data comment doi glossa
...
+    def must_be_numerical(cls, v, values, **kwargs):
+        try:
+            ident = 1.0 * v
+        except TypeError:
+            try:
+                ident = Decimal('1.0') * v
+            except TypeError:
+                raise ValueError('Datum data should be float, Decimal, or np.ndarray')
+
+        return v
```
Comment (dgasmith) [inline qcelemental/datum.py]: As LGTM mentions ident is never used, did you mean to return this?

Comment (loriab) [reply qcelemental/datum.py]: Yes, I had a fix (now pushed). Was waiting to compbine it with any other fixes spotted.

**Comment by loriab** [comment]
I was thinking that the molecule dtype=2 PR and this PR contained new fields and slightly breaking changes such that the next mint that includes them could be `v0.3.0`.

**Comment by dgasmith** [comment]
Sounds good on `v0.3`. I think this is ready to go?

**Comment by loriab** [comment]
yes, I think this and the mol dtype=2 are RTG. If you pull in one, I'll rebase the other and patch up the changelog.

**Pull request merged**